receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -289,6 +513,118 @@ func payloadToLogs(now time.Time, payload []byte) (plog.Logs, error) {
 	return logs, nil
 }
 
+type cachePutOptions struct {
+	Expires bool          `mapstructure:"expires"`
+	TTL     time.Duration `mapstructure:"ttl"`
+}
+
+// alertCache wraps a sync Map so it is goroutine safe as well as
+// can have custom marshaling
+type alertCache struct {
+	data sync.Map `mapstructure:"data"`
+}
+
+func (a *alertCache) UnmarshalJSON(data []byte) error {
+	var candidate map[string]any
+	if err := json.Unmarshal(data, &candidate); err != nil {
+		return err
+	}
+	for k, v := range candidate {
+		a.data.Store(k, v)
+	}
+	return nil
+}
+
+func (a *alertCache) MarshalJSON() ([]byte, error) {
+	candidate := make(map[string]any)
+	a.data.Range(func(key, value any) bool {
+		k, _ := key.(string)
+		candidate[k] = value
+		return true
+	})
+	return json.Marshal(candidate)
+}
+
+func (a *alertCache) Has(k string) bool {
+	val, ok := a.data.Load(k)
+	if ok {
+		v := val.(cachePutOptions)
+		now := time.Now()
+		if now.Sub(now.Add(-v.TTL)).Seconds() > v.TTL.Seconds() {
```

Comment:
I don't think there's actually enough information to determine if TTL has passed since insertion here; I think instead of a TTL duration, you probably just want a timestamp for when the record expires.